fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/MatchedOrderingPart.java

```diff
@@ -92,20 +76,18 @@ public boolean equals(final Object o) {
             return false;
         }
         final MatchedOrderingPart that = (MatchedOrderingPart)o;
-        return getOrderingPart().equals(that.getOrderingPart()) &&
-               Objects.equals(getQueryPredicate(), that.getQueryPredicate());
+        return getOrderingPart().equals(that.getOrderingPart());
     }
 
     @Override
     public int hashCode() {
-        return Objects.hash(getOrderingPart().hashCode(), getQueryPredicate());
+        return Objects.hash(getOrderingPart().hashCode());
```

Comment:
Can you add the `isReverse` bit of information here and in the `equals()`. I hope it was just an oversight and there is no hidden reason to not have it in here.

Reply:
`isReverse` is member of `OrderingPart` and it is included in calculating equality and hash code. So I think it is not necessary to have it included here again, right?

Reply:
I added `comparisonRangeType` to both equality and hash code calculation.

Reply:
Sorry, I was being blind, I meant `comparisonRangeType` but you already picked up on my mistake.

Reply:
no worries at all